File: fix_errors.py
import os
import re

def fix_initial_value(file_path):
    with open(file_path, 'r') as f:
        content = f.read()
    
    # Replace initialValue with value when it's inside a DropdownButtonFormField or entity_picker.dart
    # We look for DropdownButtonFormField followed by some content and then initialValue:
    # This is a bit tricky with regex, so we'll do it more carefully.
    
    # Pattern to find DropdownButtonFormField and the next initialValue
    # This regex attempts to find DropdownButtonFormField and then the first occurrence of initialValue:
    # until it hits a closing parenthesis or next widget.
    
    # A simpler approach: if a file contains DropdownButtonFormField, 
    # and we find 'initialValue:' we can check the context.
    
    if 'DropdownButtonFormField' in content or 'EntityPicker' in content:
        # Regex to find initialValue: and check if it's likely a DropdownButtonFormField
        # We'll use a simple replacement for now and verify.
        # Actually, let's use a more surgical approach.
        
        lines = content.split('\n')
        new_lines = []
        is_inside_dropdown = False
        for line in lines:
            if 'DropdownButtonFormField' in line:
                is_inside_dropdown = True
            
            if is_inside_dropdown and 'initialValue:' in line:
                line = line.replace('initialValue:', 'value:')
            
            if is_inside_dropdown and ');' in line: # End of widget call (approximate)
                # This is weak but might work for most cases.
                pass 
                
            new_lines.append(line)
        
        # Reset is_inside_dropdown is hard. 
        # Let's just replace all 'initialValue:' with 'value:' if it's NOT a 'TextFormField'.
        # Wait, that's also risky.
        
        # How about: find all occurrences of DropdownButtonFormField and replace the next initialValue
        content = re.sub(r'(DropdownButtonFormField<[^>]*>\s*\(\s*)initialValue:', r'\1value:', content)
        content = re.sub(r'(DropdownButtonFormField\s*\(\s*)initialValue:', r'\1value:', content)
        
        # Also for cases where it's not immediately after (more common)
        # We'll match DropdownButtonFormField and then some content until initialValue:
        # We need to be careful not to skip over other widgets.
        
        # Let's use a more robust regex:
        # Match DropdownButtonFormField, then any characters except 'TextFormField' or 'TextEditingController' 
        # until initialValue:
        def replace_dropdown(match):
            return match.group(0).replace('initialValue:', 'value:')

        content = re.sub(r'DropdownButtonFormField[^{};]*?initialValue:', replace_dropdown, content, flags=re.DOTALL)
        
        # Special case for entity_picker.dart
        if 'entity_picker.dart' in file_path:
             content = content.replace('initialValue: widget.value', 'value: widget.value')

    with open(file_path, 'w') as f:
        f.write(content)

for root, dirs, files in os.walk('lib'):
    for file in files:
        if file.endswith('.dart'):
            fix_initial_value(os.path.join(root, file))

File: fix_errors.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            results = results.concat(walk(file));
        } else {
            results.push(file);
        }
    });
    return results;
}

function fixInitialValue(filePath) {
    if (!filePath.endsWith('.dart')) return;
    
    let content = fs.readFileSync(filePath, 'utf8');
    
    if (content.includes('DropdownButtonFormField') || content.includes('EntityPicker')) {
        // Replace initialValue: with value: in DropdownButtonFormField context
        // We use a regex that looks for DropdownButtonFormField followed by some content and then initialValue:
        // We match until a closing parenthesis ); or another widget definition.
        
        // This regex is slightly different in JS. We use the 's' flag for dotAll if available, or [^]
        content = content.replace(/DropdownButtonFormField[^]*?initialValue:/g, (match) => {
            // Ensure we didn't cross into another widget like TextFormField
            if (match.includes('TextFormField')) {
                return match; // Don't replace if there's a TextFormField in between
            }
            return match.replace('initialValue:', 'value:');
        });

        if (filePath.endsWith('entity_picker.dart')) {
            content = content.replace('initialValue: widget.value', 'value: widget.value');
        }
    }
    
    fs.writeFileSync(filePath, content);
}

const files = walk('lib');
files.forEach(fixInitialValue);
